WordPress/src/main/java/org/wordpress/android/ui/accounts/login/Login2FaFragment.java

```diff
@@ -54,13 +71,46 @@
     ArrayList<Integer> mOldSitesIDs;
 
     private String mEmailAddress;
+    private String mIdToken;
+    private String mNonce;
+    private String mNonceAuthenticator;
+    private String mNonceBackup;
+    private String mNonceSms;
     private String mPassword;
-
-    public static Login2FaFragment newInstance(String emailAddress, String password) {
+    private String mService;
+    private String mType;
```

Comment:
The `mType` cannot be deduced from the _nonce_ length.  The `mType` is determined by the _code_ length.

The `mType` doesn't need to be persisted on device rotation since it is set with `setAuthCodeTypeAndNonce` before passing it to `PushSocialAuthPayload`.  I can remove it from `onCreate` and remove the `ARG_2FA_TYPE` constant.

I didn't want to put that level of logic in FluxC.  The network layer should not care about app-level logic.  It should just take parameters, make network calls, and return responses.

Reply:
In principal yes, the networking lib should only handle network stuff and not that much business logic, but FluxC is not just a networking library by any measure. Instead, it tries to hide away many WP/WPCOM details, exposing simple and consistent interfaces to `wpandroid` and others.

In this case, the 2FA type looks like noise that the mobile client shouldn't even know/care about at all. Unless it serves some purpose to have it exposed to the user, we should hide it inside FluxC (if possible). But, this is not something this PR needs to block on anyway. So, let's keep it as is but, I would still recommend opening a ticket in FluxC to hide the 2FA type from the caller. Thanks!

Reply:
I removed the unnecessary code for two-factor authentication type in 40c1b33dd2e0c778b9a5d76491e4119d3ec28654.

Reply:
Oh, I think we overdid it here. `mType` doesn't need to be initialised from fragment's argument but we _do_ need to preserve it on rotation so its value is correct when the `onSocialChanged` response arrives. Rotation may have happened after the `PushSocialAuthPayload` call but before the response comes back.

Reply:
I added state preservation for `mType` in b71176d000c892d5f39a73167a560001f8121d96.